service/csi_extension_server.go

```diff
@@ -434,3 +435,14 @@ func (s *service) buildCreateVGSResponse(ctx context.Context, snapResponse *siot
 	return groupSnapshots, nil
 
 }
+
+func (s *service) ParseVolumeHandle(ctx context.Context, request *volumeGroupSnapshot.VolumeHandleRequest) (*volumeGroupSnapshot.VolumeHandleResponse, error) {
+	if request.VolumeHandle == "" {
+		return nil, errors.New("volume handle should not be empty")
+	}
+	volHandle := strings.Split(request.VolumeHandle, "-")
```

Comment:
We used similar logic and ran into issues where ArrayID was actually a system name with hyphens.
@nb950, is that a possible issue here as well?

Could implement something similar to [this](https://github.com/dell/csi-powerflex/blob/be5bedda3fd00f29cc1238fbe69b22f6c180991e/service/service.go#L664)

Reply:
@JacobGros This is the similar thing done as of now in the sidecar, I just took the same thing and added it in this parser function https://github.com/dell/csi-volumegroup-snapshotter/blob/main/controllers/dellcsivolumegroupsnapshot_controller.go#L861

Reply:
Probably inside this parser RPC we can just call  `getVolumeIDFromCsiVolumeID` and ` getSystemIDFromCsiVolumeID`  methods by passing the entire volume handle. @nb950 @JacobGros Please let me know if this works. Thank you

Reply:
sounds good to me

Reply:
Done